### ISSUE TITLE
Optional random number generation for get_params methods in image transforms

### ISSUE THREAD

**Issue by danielhauagge** [pytorch/vision#638]
In order to make the process of mini batch generation more deterministic, it would be nice if the `get_params` method could take as an optional parameter a random number generator.

**Comment by fmassa** [comment]
@danielhauagge I think this is a reasonable request.
But at the same time, we can also set the random seed in our implementation, which makes it also deterministic, so I'm not sure it would be worth the trouble.

Thoughts?

**Comment by danielhauagge** [comment]
@fmassa not exactly sure what you mean. Are you talking about having some function that takes as input a random seed and creates a random number generator that is shared by all transforms? If so, then I think that would achieve the same effect for my purposes.

**Comment by fmassa** [comment]
@danielhauagge I mean that we can set the random seed in our datasets `__getitem__`, or via a custom transform.
For example
```python
class FixSeed(object):
    def __init__(self, seed):
        self.seed = seed
    def __call__(self, *args, **kwargs):
        torch.manual_seed(self.seed)
```

Does this solve your issue?

**Comment by danielhauagge** [comment]
If I understand correctly, in your example `FixSeed` is a generic transform class. Then what would happen is that, for every call, I'd get the same output. Shouldn't the call to `manual_seed` be in the constructor? If so, then I think this could work. I'm not sure how random number generation works in torch, but wouldn't you also need to hold a reference to a random number generator, to avoid having one call to `manual_seed` interfere with global random number generator?

```python
class FixSeed(object):
    def __init__(self, seed=None):
        if seed:
            self._rng = create_a_random_number_generator_with_seed(seed)
       else:
           self._rng = get_global_random_number_generator()
```

**Comment by fmassa** [comment]
I'm not sure I follow: when do you want to fix the seed, once for the whole training?
In this case, can't you do that in the main script, and pass the `init_fn` to your dataloader?
